torchao/csrc/cuda/fp6_llm/utils_core.cuh

```diff
@@ -98,13 +98,13 @@ __device__ __forceinline__ void core_mma_slice(float                     c[][REG
     #pragma unroll
     for (int i = 0; i < WARP_ROW_MMA_TENSORS; i++) {
         if(TilingConfig::WARP_COL_MMA_TENSORS==1) {
-            MMA_FP16_M16N8K16( c_uint_ptr[i], a_read[i], b_read[0] );
+            MMA_FP16_M16N8K16<USE_BF16>( c_uint_ptr[i], a_read[i], b_read[0] );
```

Comment:
more of a nit: but USE_BF16 is now being piped through the entire codebase when that feels like it can be inferred from the type of an argument instead

Reply:
I get your point, but I'm not sure this can be avoided. The point is that for many of the functions (such as `core_mma_slice` shown above), the array input types don't indicate whether float16 or bfloat16 is used (e.g. using generic `uint32_t` types). If we derive the type from the input type, then the input type needs to be templated (which is what I started out with), but leads to a similar effect where you have to pipe the template type through the entire codebase. I've made an effort to only pass this `USE_BF16` parameter when it's strictly necessary.